src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -73,9 +84,22 @@ static int osrandom_init(ENGINE *e) {
 
 static int osrandom_rand_bytes(unsigned char *buffer, int size) {
     ssize_t n;
+    struct stat st;
+
+    if (fstat(urandom_cache.fd, &st)
+        || st.st_dev != urandom_cache.st_dev
+        || st.st_ino != urandom_cache.st_ino) {
+        /* Something changed: forget the cached fd (but don't close it,
+           since it probably points to something important for some
+           third-party code). Re-initialize the engine. */
+        int res = osrandom_init(NULL);
```

Comment:
I'd be okay with erroring out here rather than trying to recover. We could also push an error onto the OpenSSL stack so we might have a chance of figuring out why this is happening in the future when some bad software closes all fds on us.